File: 1-3/solution.cpp
#include<iostream>
#include<string>


using namespace std;

string URLify(string s, int size)
{
	int space=0;
	for(int i=0;i<size; i++){
		if(s[i]==' ')
			space++;
	}
	
	int target=size+2*space-1;
	
	for(int i=size-1;i>=0;i--){
		if(s[i]!=' ')
		{
			s[target]=s[i];
			target--;
		}
		else
		{
			s[target]='0';
			s[target-1]='2';
			s[target-2]='%';
			target=target-3;
		}
					
	}
	return s;	

}

int main()
{

	string s = "My name Mayank    ";
	int size = 14;
	cout<< "Augmented string is : " << URLify(s, size) << endl;
	return 0;
}

File: 1-3/solution.py
#!/usr/bin/python

def URLify(s,size):
	space=0
	for i in range(size):
		if s[i]==' ':
			space=space+1

	target=size+2*space-1
	for i in range(size-1,-1,-1):
		if s[i]==' ':
			s[target-2:target+1]='%20'			# Not working, Python string is immutable
			target=target-3		
		else:
			s[target]=s[i]						# Not working, Python string is immutable
			target=target-1
	return s


def URLify2(s,size):
	st=''
	for i in range(size):
		if s[i]==' ':
			st+='%20'
		else:
			st+=s[i]
	return st



s="my name Mayank    "
size=14

print URLify2(s,size)
